implementations/elixir/ockam/ockam_kafka/lib/interceptor/inlet_manager.ex

```diff
@@ -0,0 +1,124 @@
+defmodule Ockam.Kafka.Interceptor.InletManager do
+  @moduledoc """
+  Dynamic inlet manager for kafka interceptor.
+
+  Inlets are GenServers and cannot be managed by ockam registry.
+  This module can dynamically create and stop inlets using base port and port offset.
+  """
+  use GenServer
+
+  def start_link([base_port, allowed_ports, base_route, outlet_prefix]) do
+    GenServer.start_link(
+      __MODULE__,
+      [base_port, allowed_ports, base_route, outlet_prefix],
+      ## TODO: make this optional
+      name: __MODULE__
+    )
+  end
+
+  @impl true
+  def init([base_port, allowed_ports, base_route, outlet_prefix]) do
+    {:ok,
+     %{
+       base_port: base_port,
+       allowed_ports: allowed_ports,
+       base_route: base_route,
+       outlet_prefix: outlet_prefix,
+       inlets: %{}
+     }}
+  end
+
+  def list_inlets(server \\ __MODULE__, timeout \\ 5000) do
+    GenServer.call(server, :list_inlets, timeout)
+  end
+
+  def set_inlets(server \\ __MODULE__, inlets, timeout \\ 5000) when is_list(inlets) do
+    GenServer.call(server, {:set_inlets, inlets}, timeout)
+  end
+
+  @impl true
+  def handle_call(:list_inlets, _from, %{inlets: inlets} = state) do
+    {:reply, inlets, state}
+  end
+
+  def handle_call(
+        {:set_inlets, requested_inlets},
+        _from,
+        %{allowed_ports: allowed_ports, inlets: inlets} = state
+      ) do
+    case Enum.any?(requested_inlets, fn port_offset -> port_offset > allowed_ports end) do
```

Comment:
From the provider inline docs:
```
base_port: optional<integer>, default 9001 - port from which to start dynamic inlets
allowed_ports: optional<integer>, default 20 - number of ports allowed for dynamic inlets (adds to base_port)
```
This does not exactly fit the implementation. The implementation assumes (but doesn't enforce) that the `requested_inlets` will all be positive and therefore in such case no inlet will be started on the base port. If base port is `9001` then we would start on `9002`.

But I think the bigger issue here that we don't check the input, so nothing stops a client to specify requested inlets using negative numbers and 0, so the allowed ports number can be easily exceeded (and any port <= base_port + allowed_ports can be used).

Reply:
> The implementation assumes (but doesn't enforce) that the requested_inlets will all be positive and therefore in such case no inlet will be started on the base port

I'm not sure why would you think so. 0 is a totally valid port offset.

> But I think the bigger issue here that we don't check the input, so nothing stops a client to specify requested inlets using negative numbers

WIll add the check.

Reply:
> I'm not sure why would you think so. 0 is a totally valid port offset.

I agree that 0 can be a valid offset, but if we allow it, then the math with the `port_offset > allowed_ports` condition doesn't check out, because 0..20 is 21. That's why I assumed from the implementation that we wanted offset to be positive.